Chore/update react version

rike422 opened the pull request:
Pull requests for the following issues

https://github.com/whitphx/react-ymd-date-select/issues/82

whitphx commented on package.json:
```diff
@@ -74,8 +74,8 @@
   "peerDependencies": {
-    "react": "^16.9.0 || ^17",
-    "react-dom": "^16.9.0 || ^17"
+    "react": "^18",
```
Is it ok to set something like `^16.9.0 || ^17 || ^18` for the backward compatibility?

rike422 replied:
I think you are right. There is nothing wrong with the source code so there was no need to remove support.
I have modified the code as you pointed out.

whitphx reviewed (commented):
Thanks!

whitphx reviewed (commented):
Thanks, LGTM!
Then can you check the error at `yarn build`?

rike422 commented:
@whitphx 

Thanks for the review.

## formik

formik maintenance is not active. Fixed a type error that occurred when using react18.
Please refer to the following ticket for more information.

https://github.com/jaredpalmer/formik/issues/3546

## react-live-editor

Fixed a type error when using the editor.

It seems to have occurred because react-simple-code-editor moved from flowtype to typescript in version 0.11.1.

Default values were defined using the following code.

https://github.com/satya164/react-simple-code-editor/blob/v0.11.3/src/index.tsx#L106

whitphx reviewed (approved):
LGTM!

Thank you for your contribution while the change got bigger than expected.
